### PR TITLE
update to duckdb 0.6

### PR DESCRIPTION
**Pull request by changhiskhan**

### CODE REVIEW

File: integration/duckdb/include/ml/catalog.h [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-//  limitations under the License.
 
 #pragma once
 
-#include <duckdb.hpp>
 #include <map>
 #include <memory>
 #include <mutex>
 
+#include "duckdb.hpp"
```
Comment (eddyxu) [inline integration/duckdb/include/ml/catalog.h:22]: why changed `duckdb.hpp` to a local header here?

File: integration/duckdb/include/ml/pytorch.h [hunk 2]
```diff
@@ -23,7 +24,7 @@
 #include <opencv2/opencv.hpp>
 #include <vector>
 
-#include "lance/duckdb/ml/catalog.h"
+#include "catalog.h"
```
Comment (eddyxu) [inline integration/duckdb/include/ml/pytorch.h:27]: let's use the full path as include path, i.e., `ml/catelog.h`?

File: integration/duckdb/src/lance/duckdb/lance-extension.cc [hunk 3]
```diff
@@ -14,47 +14,49 @@
...
 
+#include "lance-extension.h"
+
 #include <duckdb.hpp>
 
-#include "lance/duckdb/list_functions.h"
-#include "lance/duckdb/ml/functions.h"
-#include "lance/duckdb/vector_functions.h"
+#include "list_functions.h"
+#include "vector_functions.h"
```
Comment (eddyxu) [inline integration/duckdb/src/lance/duckdb/lance-extension.cc:22]: Do we plan to expose duckdb extension c++ code to others?
Should we maintain the namespace if so?

File: integration/duckdb/include/ml/catalog.h (hunk 1 above)
Comment (changhiskhan) [reply integration/duckdb/include/ml/catalog.h:22]: copy paste error while i was refactoring

**Review by eddyxu — APPROVED** [review]
LGTM

Pending CI

**Pull request merged**